src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -118,6 +119,11 @@ class ProtobufPlugin implements Plugin<Project> {
           // block. Only at this point the configuration has been finalized.
           project.protobuf.tools.registerTaskDependencies(project.protobuf.getGenerateProtoTasks().all())
         }
+
+        project.plugins.withType(IdeaPlugin) { IdeaPlugin ip ->
+            ip.model.module.sourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+            ip.model.module.generatedSourceDirs += project.file("${project.protobuf.generatedFilesBaseDir}/main/java");
+        }
```

Comment:
Without `generatedSourceDirs` the folder is not marked as *generated* but as normal source directory.

With `generatedSourceDirs` alone the folder isn't recognized at all (maybe because it's nested inside an already excluded folder)